doppler/resource_secret.go

```diff
@@ -92,6 +113,21 @@ func resourceSecretRead(ctx context.Context, d *schema.ResourceData, m interface
 		return handleNotFoundError(err, d)
 	}
 
+	nilFields := []string{}
+	if secret.Value.Raw == nil {
+		nilFields = append(nilFields, "raw")
+	}
+	if secret.Value.Computed == nil {
+		nilFields = append(nilFields, "computed")
+	}
+
+	if len(nilFields) > 0 {
+		return diag.FromErr(fmt.Errorf(
+			"One or more secret fields are restricted: %v. "+
+				"Make sure that you are using a service account or service token to perform Terraform actions. "+
+				"Otherwise, Terraform won't be able to fetch restricted secrets to check the validity of its state.", nilFields))
```

Comment:
optional:
```suggestion
				"You must use a service account or service token to manage these resources. "+
				"Otherwise, Terraform cannot fetch these restricted secrets to check the validity of their state.", nilFields))
```